src/3d/qgs3drendercontext.h

```diff
@@ -0,0 +1,175 @@
+/***************************************************************************
+  qgs3drendercontext.h
+  --------------------------------------
+  Date                 : August 2024
+  Copyright            : (C) 2024 by Nyall Dawson
+  Email                : nyall dot dawson at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGS3DRENDERCONTEXT_H
+#define QGS3DRENDERCONTEXT_H
+
+#include "qgis_3d.h"
+#include "qgsvector3d.h"
+#include "qgsrectangle.h"
+#include "qgsrange.h"
+#include "qgscoordinatereferencesystem.h"
+#include "qgscoordinatetransformcontext.h"
+#include "qgsexpressioncontext.h"
+
+#include <QColor>
+
+class QgsTerrainGenerator;
+class Qgs3DMapSettings;
+
+#define SIP_NO_FILE
+
+/**
+ * \ingroup 3d
+ * \brief Rendering context for preparation of 3D entities.
+ *
+ * Contains a snapshot of properties from Qgs3DMapSettings, in a thread-safe, cheap-to-copy structure.
```

Comment:
Can you add a comment to explain that no QObject are allowed in this class?

Reply:
Done